module/zfs/zfs_vnops.c

```diff
@@ -4090,8 +4090,9 @@ zfs_putpage(struct inode *ip, struct page *pp, struct writeback_control *wbc)
 
 	err = sa_bulk_update(zp->z_sa_hdl, bulk, cnt, tx);
 
-	zfs_log_write(zsb->z_log, tx, TX_WRITE, zp, pgoff, pglen, 0,
-	    zfs_putpage_commit_cb, pp);
+	if (zsb->z_log != NULL)
+		zfs_log_write(zsb->z_log, tx, TX_WRITE, zp, pgoff, pglen, 0,
+		    zfs_putpage_commit_cb, pp);
```

Comment:
This should never be a possible case in the existing code.   Pages which are part of a snapshot should never be allowed to be dirtied, see commit 1e8db7710220332808920a582e5794d6fc37b109.  This check was added just in case somehow a page got dirtied.  Now that pretty clearly can never happen or we would have NULL dereferenced here long ago and there are no issues like this open in the tracker.  Given that I'd rather remove the NULL check around `zil_commit()` a few lines down.

This also has the advantage of bringing this code slightly closer inline with upstream at least in terms of intent and assumptions.

@tuxoko what are your thoughts on this.